### PR TITLE
Enhance test summary handling

### PR DESCRIPTION
**Pull request by liangxia**

### CODE REVIEW

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-aws-ipi-ovn-ipsec-to-multiarch-f2

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-aws-ipi-ovn-ipsec-to-multiarch-f2

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-aws-ipi-ovn-ipsec-to-multiarch-f2

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-aws-ipi-ovn-ipsec-to-multiarch-f2

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-aws-ipi-ovn-ipsec-to-multiarch-f2

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-aws-ipi-ovn-ipsec-to-multiarch-f2

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-aws-ipi-ovn-ipsec-to-multiarch-f2

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-aws-ipi-ovn-ipsec-to-multiarch-f2

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-aws-ipi-ovn-ipsec-to-multiarch-f2

**Comment by liangxia** [comment]
Pass [log](https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/54237/rehearse-54237-periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-aws-ipi-ovn-ipsec-to-multiarch-f2/1812705540879421440/artifacts/aws-ipi-ovn-ipsec-to-multiarch-f2/cucushift-upgrade-prepare/build-log.txt)

/cc @JianLi-RH @dis016 @pruan-rht @jianlinliu

File: ci-operator/step-registry/cucushift/upgrade/check/cucushift-upgrade-check-commands.sh [hunk 1]
```diff
@@ -254,7 +254,18 @@ function test_execution() {
 function summarize_test_results() {
     # summarize test results
     echo "Summarizing test results..."
-    [[ -e "${ARTIFACT_DIR}" ]] || exit 0
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/upgrade/check/cucushift-upgrade-check-commands.sh:257]: Shall we update ci-operator/step-registry/cucushift/chainupgrade/toimage/cucushift-chainupgrade-toimage-commands.sh together ?

Comment (liangxia) [reply ci-operator/step-registry/cucushift/upgrade/check/cucushift-upgrade-check-commands.sh:257]: Done

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-stable-aws-rosa-hcp-stage-critical-f14 periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-stable-aws-rosa-hcp-sector-byo-kms-oidc-auditlog-prod-critical-f7 periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-baremetalds-ipi-ovn-ipv4-fips-f14-wrs periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-vsphere-ipi-amd-f7-netobserv

File: ci-operator/step-registry/openshift-extended/web-tests/openshift-extended-web-tests-commands.sh [hunk 2]
```diff
@@ -52,9 +52,20 @@ fi
...
+    ls -lR "${ARTIFACT_DIR}"
+    files="$(find "${ARTIFACT_DIR}" -name '*.xml' | wc -l)"
+    if [[ "$files" -eq 0 ]] ; then
+        echo "There are no JUnit files"
+        exit 0
+    fi
+fi
 declare -A results=([failures]='0' [errors]='0' [skipped]='0' [tests]='0')
-grep -r -E -h -o 'testsuite.*tests="[0-9]+"[^>]+' "${ARTIFACT_DIR}/gui_test_screenshots/console-cypress.xml" 2>/dev/null > /tmp/zzz-tmp.log
+grep -r -E -h -o 'testsuite.*tests="[0-9]+"[^>]*' "${ARTIFACT_DIR}/gui_test_screenshots/console-cypress.xml" 2>/dev/null > /tmp/zzz-tmp.log || exit 0
```
Comment (liangxia) [inline ci-operator/step-registry/openshift-extended/web-tests/openshift-extended-web-tests-commands.sh:68]: Change from `[^>]+` to `[^>]*` because `tests="123"` can exist at the end of the line.
Also add `exit 0` so that if the grep command can not find the pattern ( then we don't know how to parse the result), exit successfully, so that the other tests following this step can move on.

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-stable-aws-rosa-hcp-stage-critical-f14 periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-stable-aws-rosa-hcp-sector-byo-kms-oidc-auditlog-prod-critical-f7 periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-baremetalds-ipi-ovn-ipv4-fips-f14-wrs periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-vsphere-ipi-amd-f7-netobserv

**Comment by liangxia** [comment]
Fix two more corner cases.  The PR is ready for final test/review.

**Comment by Amoghrd** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-multi-nightly-gcp-ipi-amd-f7-netobserv
